schema.sql

```diff
@@ -0,0 +1,68 @@
+CREATE DATABASE CatalogOfThings;
+USE CatalogOfMyThings;
+
+CREATE TABLE item (
+    id INT PRIMARY KEY,
+    title VARCHAR(255),
+    publish_date DATE,
+    publisher VARCHAR(255),
+    cover_state VARCHAR(255)
+    archived BOOLEAN,
+    label_id INT, -- Add a foreign key for the label (one-to-many relationship)
+    genre_id INT, -- Add a foreign key for the genre (one-to-many relationship)
+    author_id INT, -- Add a foreign key for the author (one-to-many relationship)
+    FOREIGN KEY (label_id) REFERENCES labels (id)  
+    FOREIGN KEY (genre_id) REFERENCES genres (id),
+    FOREIGN KEY (author_id) REFERENCES authors (id),
+);
+
+--create music album table--
+CREATE TABLE MUSIC_ALBUM(
+    ID INTEGER PRIMARY KEY,
+    ON_SPOTIFY BOOLEAN,
+    ARCHIVED BOOLEAN,
+    PUBLISH_DATE DATE,
+    GENRE VARCHAR(100) REFERENCES GENRE(NAME),
+    AUTHOR VARCHAR(100) REFERENCES authors(first_name),
+    LABEL VARCHAR(100) REFERENCES labels(TITLE),
+)
+
+--Create Genre table--
+CREATE TABLE IF NOT EXISTS GENRE(
+    ID INTEGER,
+    NAME CHAR(32) NOT NULL,
+    ITEMS INT REFERENCES (books, MUSIC_ALBUM, games),
+    PRIMARY KEY(ID, items)
+)
+
+-- Create the games table
+CREATE TABLE games (
+  id INT PRIMARY KEY,
+  item_id INT, -- A foreign key to reference the parent item table
+  title VARCHAR(255),
+  publish_date DATE,
+  multiplayer BOOLEAN,
+  last_played_at DATE,
+  archived BOOLEAN DEFAULT false,
+  author_id INT, -- Add a foreign key for the author (one-to-many relationship)
+  FOREIGN KEY (author_id) REFERENCES authors (id)
```

Comment:
- [x] The `games` table should have only 3 columns `multiplayer`, `last_played_at`, and `id`(which is referencing the item id). Please fix that to match the given [UML diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png)

![image](https://github.com/Uzair-Manzoor/RubyCapstone/assets/68796232/7a97864a-b8a1-4c37-b2a5-ee34ba6c234c)

```suggestion
CREATE TABLE game(
	id INT NOT NULL,
	multiplayer BOOLEAN,
	last_played_at DATE,
	FOREIGN KEY (id) REFERENCES item (id)
);
```

Reply:
- Well resolved team